src/Action/ExceptionAction.php

```diff
@@ -66,6 +66,8 @@ public function __construct(SerializerInterface $serializer, array $errorFormats
     public function __invoke(FlattenException $exception, Request $request): Response
     {
         $exceptionClass = $exception->getClass();
+        $statusCode = $exception->getStatusCode();
+
         foreach ($this->exceptionToStatus as $class => $status) {
             if (is_a($exceptionClass, $class, true)) {
                 $exception->setStatusCode($status);
```

Comment:
You have to change the value of your `$statusCode` var here to treat the `FlattenException` as immutable.

```diff
- $exception->setStatusCode($status);
+ $statusCode = $status;
```